cognite/neat/rules/exporters/_loaders.py

```diff
@@ -0,0 +1,261 @@
+            local_dumped.pop("properties", None)
+
+        return local_dumped == cdf_resource_dumped
+
+    def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, View]) -> list[View]:
+        """Retrieves all ancestors of a view.
+
+        This will mutate the cache passed in, and return a list of views that are the ancestors
+        of the views in the parents list.
+
+        Args:
+            parents: The parents of the view to retrieve all ancestors for
+            cache: The cache to store the views in
+        """
+        parent_ids = parents
+        found: list[View] = []
+        while parent_ids:
+            to_lookup = []
+            grand_parent_ids = []
+            for parent in parent_ids:
+                if parent in cache:
+                    found.append(cache[parent])
+                    grand_parent_ids.extend(cache[parent].implements or [])
+                else:
+                    to_lookup.append(parent)
+
+            if to_lookup:
+                looked_up = self.client.data_modeling.views.retrieve(to_lookup)
+                cache.update({view.as_id(): view for view in looked_up})
+                found.extend(looked_up)
+                for view in looked_up:
+                    grand_parent_ids.extend(view.implements or [])
+
+            parent_ids = grand_parent_ids
+        return found
+
+
+class ContainerLoader(DataModelingLoader[ContainerId, ContainerApply, Container, ContainerApplyList, ContainerList]):
+    resource_name = "containers"
+
+    @classmethod
+    def get_id(cls, item: Container | ContainerApply) -> ContainerId:
+        return item.as_id()
+
+    def sort_by_dependencies(self, items: Sequence[ContainerApply]) -> list[ContainerApply]:
+        container_by_id = {container.as_id(): container for container in items}
+        container_dependencies = {
+            container.as_id(): {
+                const.require
+                for const in container.constraints.values()
+                if isinstance(const, RequiresConstraint) and const.require in container_by_id
+            }
+            for container in items
+        }
+        return [
+            container_by_id[container_id] for container_id in TopologicalSorter(container_dependencies).static_order()
+        ]
+
+    def create(self, items: Sequence[ContainerApply]) -> ContainerList:
+        return self.client.data_modeling.containers.apply(items)
+
+    def retrieve(self, ids: SequenceNotStr[ContainerId]) -> ContainerList:
+        return self.client.data_modeling.containers.retrieve(cast(Sequence, ids))
+
+    def update(self, items: Sequence[ContainerApply]) -> ContainerList:
+        return self.create(items)
+
+    def delete(self, ids: SequenceNotStr[ContainerId]) -> list[ContainerId]:
+        return self.client.data_modeling.containers.delete(cast(Sequence, ids))
+
+
+class DataModelLoader(DataModelingLoader[DataModelId, DataModelApply, DataModel, DataModelApplyList, DataModelList]):
```

Comment:
Create `cognite/neat/rules/exporters/_dms/` and bundle dms specific things together. We use this approach for owl importer. Keeps things nice and clean